server/container_create_linux.go

```diff
@@ -330,17 +331,74 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 			specgen.AddMount(mnt)
 		}
 	}
-	mountLabel := sb.MountLabel()
-	processLabel := sb.ProcessLabel()
-	selinuxConfig := containerConfig.GetLinux().GetSecurityContext().GetSelinuxOptions()
-	if selinuxConfig != nil {
-		var err error
-		processLabel, mountLabel, err = getSELinuxLabels(selinuxConfig, privileged)
-		if err != nil {
+	imageSpec := containerConfig.GetImage()
+	if imageSpec == nil {
+		return nil, fmt.Errorf("CreateContainerRequest.ContainerConfig.Image is nil")
+	}
+
+	image := imageSpec.Image
+	if image == "" {
+		return nil, fmt.Errorf("CreateContainerRequest.ContainerConfig.Image.Image is empty")
+	}
+	images, err := s.StorageImageServer().ResolveNames(image)
+	if err != nil {
+		if err == storage.ErrCannotParseImageID {
+			images = append(images, image)
+		} else {
 			return nil, err
 		}
 	}
 
+	// Get imageName and imageRef that are later requested in container status
+	var (
+		imgResult    *storage.ImageResult
+		imgResultErr error
+	)
+	for _, img := range images {
+		imgResult, imgResultErr = s.StorageImageServer().ImageStatus(s.ImageContext(), img)
+		if imgResultErr == nil {
+			break
+		}
+	}
+	if imgResultErr != nil {
+		return nil, imgResultErr
+	}
+
+	var labelOptions []string
+	selinuxConfig := containerConfig.GetLinux().GetSecurityContext().GetSelinuxOptions()
+	if selinuxConfig == nil {
+		labelOptions = label.DupSecOpt(sb.ProcessLabel())
```

Comment:
why are we calling DupSecOpt, can't we just do `[]string{sb.ProcessLabel()}`? we won't change anything in the way we were doing it before this PR

Reply:
We are handing LabelOpts to CreateContainer.  DupSecOpt, tells InitSELinux to give me matching SELinux labels.